packages/core/src/account/kernel.ts

```diff
@@ -0,0 +1,119 @@
+import type {Address} from "abitype";
+import {
+    concatHex,
+    encodeAbiParameters,
+    encodeFunctionData,
+    type FallbackTransport,
+    type Hex,
+    type Transport,
+} from "viem";
+import {type BaseSmartAccountParams, BaseSmartContractAccount,} from "./base.js";
+import {SmartAccountSigner} from "../interfaces/signer";
+import {BaseValidator, ValidatorMode} from "../validator/kernel/base";
+import {KernelAccountAbi} from "../abis/KernelAccountAbi";
+import {parseAbiParameters} from "abitype";
+import {KernelFactoryAbi} from "../abis/KernelFactoryAbi";
+
+export interface KernelSmartAccountParams<
+    TTransport extends Transport | FallbackTransport = Transport
+> extends BaseSmartAccountParams<TTransport> {
+    owner: SmartAccountSigner;
+    factoryAddress: Address;
+    index?: bigint;
+    defaultValidator: BaseValidator
+    validator?: BaseValidator
+}
+
+export class KernelSmartContractAccount<
+    TTransport extends Transport | FallbackTransport = Transport
+> extends BaseSmartContractAccount<TTransport> {
+    private owner: SmartAccountSigner;
+    private readonly factoryAddress: Address;
+    private readonly index: bigint;
+
+    private defaultValidator: BaseValidator;
+    private validator: BaseValidator;
+
+
+    constructor(params: KernelSmartAccountParams) {
+        super(params);
+        this.index = params.index ?? 0n;
+        this.owner = params.owner;
+        this.factoryAddress = params.factoryAddress;
+        this.defaultValidator = params.defaultValidator!
+        this.validator = params.validator ?? params.defaultValidator!
+    }
+
+    getDummySignature(): Hex {
+        return "0x4046ab7d9c387d7a5ef5ca0777eded29767fd9863048946d35b3042d2f7458ff7c62ade2903503e15973a63a296313eab15b964a18d79f4b06c8c01c7028143c1c";
+    }
+
+    async encodeExecute(target: Hex, value: bigint, data: Hex): Promise<Hex> {
+        if (this.validator.mode !== ValidatorMode.sudo) {
+            throw new Error("Validator Mode not supported")
+        } else {
+            return this.encodeExecuteAction(target, value, data, 0n)
+        }
+    }
+
+    async encodeExecuteDelegate (target: Hex, value: bigint, data: Hex): Promise<Hex> {
+        return this.encodeExecuteAction(target, value, data, 1n)
+    }
+
+    async signWithEip6492(msg: string | Uint8Array | Hex): Promise<Hex> {
```

Comment:
I'm actually not super familiar with 6492, so this might be a dumb question. Should this not just be the default `signMessage` implementation below rather than having 2 different sign message functions? I imagine the factory supports the 6492 since it's being used to sign here, so there shouldn't be any issues on account creation right?

Reply:
the entrypoint contract doesn't accept it yet